### PR TITLE
bsunanda:Run2-alca57 Add 2 new condition objects for HCAL: SiPMParameter and SiPMCharacteristics

### PR DESCRIPTION
**Pull request by bsunanda**
Two condition objects needed for 2017 simulation/reconstruction

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@smuzaffar @davidlange6 The tests are failing because of DAS error in one example. It is happening in all PR's. Can this be fixed?

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: CondFormats/HcalObjects/interface/HcalSiPMParameter.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class HcalSiPMParameter {
+
+public:
+  /// get SiPM type
+  int      getType()        const {return type_;}
+  /// get fcByPE
+  float    getFCByPE()      const {return fcByPE_;}
+  /// get dark current
+  float    getDrakCurrent() const {return darkCurrent_;}
```
Comment (kpedro88) [inline CondFormats/HcalObjects/interface/HcalSiPMParameter.h]: Please fix typo

File: CondFormats/HcalObjects/src/HcalSiPMCharacteristics.cc [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+bool HcalSiPMCharacteristics::loadObject(int type, int pixels, float parLin1, 
+					 float parLin2, float parLin3, 
+					 float crossTalk, int auxi1,
+					 float auxi2) {
+  const HcalSiPMCharacteristics::PrecisionItem* item = findByType(type);
+  if (item) {
+    edm::LogWarning("HCAL") << "HcalSiPMCharacteristics::loadObject type " 
+			    << type << " already exists with pixels "
+			    << item->pixels_ << " NoLinearity parameters "
```
Comment (kpedro88) [inline CondFormats/HcalObjects/src/HcalSiPMCharacteristics.cc]: Please fix typo

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@mmusich @ggovi Could you please go through the code and see if it can be approved quickly.

**Comment by abdoulline** [comment]
@mmusich @ggovi 

Dear Marco and Giacomo, 
given quite long time needed to put in place (3 pre-releases to start actually using it) Offline DB conditions, and mandatory character of this change for 2017 MC production, could you help from your side, please?..

**Comment by abdoulline** [comment]
@mmusich @ggovi 

Dear Marco and Giacomo, 
given quite long time needed to put in place (3 pre-releases to start actually using it) Offline DB conditions, and mandatory character of this change for 2017 MC production, could you help from your side, please?..

**Comment by mmusich** [comment]
+1 
still @ggovi has the final word. 
@abdoulline, Giacomo is away till the end of next week. 
Two minor points:
   * where you did you get the metric of 3 pre-releases? technically you can upload your conditions even now.
   * Perhaps would be better to know beforehand of the PR submission which new CondFormats are needed and in which timescale.

**Comment by abdoulline** [comment]
I agree, long-term planning is a good thing, but not always possible, as 
it's not so easy to agree on the DB format in view of changing/developing 
detector (2017 HE is a kind of new detector to some extend) issues.
3-prereleases span I've got from our QIETypes implementation earlier this 
year: 1) formats without DB access 2) making preparations on DB side and 
uploading payloads/tags 3) activation of DB access and end-user 
usage activation.
Probably nowadays it cn be two-step procedure (1)+(2) and (3)? 
Usually end-user stuff needs a separate PR anyway.

**Comment by bsunanda** [comment]
@ggovi Could you review this and approve this?

**Comment by abdoulline** [comment]
@mmusich, @davidlange6

Marco has mentioned above: 
"Giacomo is away till the end of next week." 
Does it mean all the PRs marked as "db-pending" are suspened for at least 2 weeks?

**Comment by mmusich** [comment]
@abdoulline if there are no further comments, we can bring this PR for discussion at today's ORP even if db-pending.

**Comment by abdoulline** [comment]
That's would be great...

**Pull request merged**